lib/FusionInventory/Agent/Tools/Win32.pm

```diff
@@ -250,23 +250,31 @@ sub getRegistryValue {
         return _call_win32_ole_dependent_api($win32_ole_dependent_api);
     }
 
-    my $key = _getRegistryKey(
-        logger  => $params{logger},
-        root    => $root,
-        keyName => $keyName
-    );
-
-    return unless (defined($key));
-
-    if ($valueName eq '*') {
-        my %ret;
-        foreach (keys %$key) {
-            s{^/}{};
-            $ret{$_} = $params{withtype} ? [$key->GetValue($_)] : $key->{"/$_"} ;
-        }
-        return \%ret;
+    if ($root =~ m/\*\*/) {
+        return _getRegistryDynamic(
+            logger    => $params{logger},
+            root      => $root,
+            keyName   => $keyName,
+            valueName => $valueName,
+            withtype  => $params{withtype}
+        );
     } else {
```

Comment:
As your `return` from your `if` block, you don't need to put the rest of the code in a `else` block.